tests/libcxx/CMakeLists.txt

```diff
@@ -1,13 +1,46 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
-# read tests.supported, sanitize the cpp-file, and create the test-case
+# Read tests.supported, sanitize the cpp-file, and create the test-case.
 if (ENABLE_FULL_LIBCXX_TESTS)
     file(STRINGS "tests.supported" alltests)
 else()
     file(STRINGS "tests.supported.default" alltests)
 endif()
 
+# List of unsupported tests with GCC.
+set(GCC_UNSUPPORTED_LIST
```

Comment:
This list is a little surprising to me ... do we know why these fail for GCC? I would expect that GCC7 implemented pretty much all of the C++17 features already.

Reply:
All the tests associated with "deduction" failed with gcc during the compile-time. I don't know the root cause yet.

Reply:
Update:
Those test cases require certain compiler support, see

https://github.com/openenclave/openenclave/blob/2536d0cc736d7cbbbb9ab4193d37ea983ecd39d9/3rdparty/libcxx/libcxx/utils/libcxx/test/config.py#L461

According to  https://gcc.gnu.org/projects/cxx-status.html#cxx17, it is supported by gcc 8.

Reply:
I re-organized that list based on this issue.

Reply:
I reorganized that list based on this issue.